### PR TITLE
chore: document collection counts metadata & parameters

### PR DESCRIPTION
**Pull request by cmars**
Defining a `meta` object convention for representing collection counts,
and query parameters to request counts on collections.

### CODE REVIEW

File: docs/standards/rest.md [hunk 1]
```diff
@@ -197,6 +245,40 @@ The following parameters are reserved for specific purposes in our API. These na
...
+If `meta_count` is not specified but `meta_count_by` is specified, `meta_count=only` is implied. The following table represents the default behaviors of different combinations of these parameters and their resulting affect on the response contents:
+
+| `meta_count` value | `meta_count_by` value | Response has `.meta.count` | Response has `.meta.count_by` | Response has `.data[]` |
+|------------|---------------|--------|-|-|
+| _not set_ | _not set_ | ❌ | ❌ | ✅ |
+| _not set_ | _one or more group(s)_ | ✅ | ✅  | ❌ |
+| `only` | _*_ | ✅ | _*_ | ❌ |
+| `with` | _*_ |  ✅ | _*_ | ✅ |
+
+An endpoint may choose to change this default behavior. If so, it must be documented in the parameter(s) description(s).
```
Comment (mitechie) [inline docs/standards/rest.md:280]: Should this be a stronger suggestion in the standard? I'm thinking how in v1 there are some APIs that support pagination and some that don't. I almost wonder if the ideal is that some level of the resource tree should have a default meta_count functionality supported.

**Review by jcsackett — COMMENTED** [review]
This seems fine, but I'm wondering if we are having too many options where "this is the default, if you're doing something else you must document it"; maybe we need to start saying "do it this way"?

Do we have known cases already where folks have a need to do something else?

File: docs/standards/rest.md (hunk 1 above)
Comment (cmars) [reply docs/standards/rest.md:280]: Obtaining the total count and grouped counts is likely to be a separate query that has to be made, and possibly an index scan to get an exact count. For a sufficiently large collection, you might cache this, or estimate counts in a HyperLogLog structure instead (but that's another thing to then store and keep in sync). This makes me hesitant to declare it a MUST across the board for all resources up front, before we know what we're in for here.

The things we have going for us here though, that v1 lacks:

- The OpenAPI spec will still declare whether a given resource collection supports counts, and how to get it.
- Customer demand can then drive improvement in those specs which do not have counts, with new version releases.

We could decide later that API version releases after a certain date must include counts. I think that would need an ADR for the reasons above. If we have that though, and we're confident that services can handle it, I'd feel comfortable that we could roll out such a constraint for all API versions releasing after that decision date.

Comment (cmars) [reply docs/standards/rest.md:280]: It seems reasonable to make the _behavior_ standard though... "if you implement counts, this is how it works". Just wanted to separate out these different threads.

Comment (cmars) [reply docs/standards/rest.md]: Shored up the wording on this one. @jcsackett had similar concerns, I'm convinced.

**Comment by cmars** [comment]
> This seems fine, but I'm wondering if we are having too many options where "this is the default, if you're doing something else you must document it"; maybe we need to start saying "do it this way"?
> 
> Do we have known cases already where folks have a need to do something else?

I think you're right. Better to provide strong direction here -- and a rule that can be automated, than leave it too open. Updated!

**Review by jcsackett — APPROVED** [review]
Thanks for the updates @cmars!

**Pull request merged**